sprocs/group_users_insert.sql

```diff
@@ -0,0 +1,77 @@
+DROP FUNCTION IF EXISTS group_users_insert(bigint,bigint,bigint,text,text);
+CREATE OR REPLACE FUNCTION
+    group_users_insert (
+        arg_assessment_id bigint,
+        arg_user_id bigint,
+        arg_authn_user_id bigint,
+        arg_group_name text,
+        arg_join_code text
+    ) RETURNS void
+AS $$
+DECLARE
+    arg_group_id bigint;
+    arg_cur_size bigint;
+    arg_max_size bigint;
+BEGIN
+    -- find group id
+    SELECT 
+        g.id
+    INTO 
+        arg_group_id
+    FROM 
+        groups AS g
+        JOIN group_configs AS gc ON g.group_config_id = gc.id
+    WHERE 
+        g.name = arg_group_name 
+        AND g.join_code = arg_join_code
+        AND gc.assessment_id = arg_assessment_id
+        AND g.deleted_at IS NULL
+        AND gc.deleted_at IS NULL;
+
+    IF NOT FOUND THEN
+        RAISE EXCEPTION 'Cannot find the group by provided join code: %', arg_join_code;
+    END IF;
+
+    -- lock the group
```

Comment:
This is great! I really like the clear pattern of "find the group, lock it, check the size constraint, modify the group". This is very safe and easy to understand.